builder/vsphere/common/step_config_params.go

```diff
@@ -39,17 +40,22 @@ func (s *StepConfigParams) Run(_ context.Context, state multistep.StateBag) mult
 		configParams = s.Config.ConfigParams
 	}
 
-	if s.Config.ToolsSyncTime {
-		configParams["tools.syncTime"] = "TRUE"
-	}
+	var info *types.ToolsConfigInfo
+	if s.Config.ToolsSyncTime || s.Config.ToolsUpgradePolicy {
+		info = &types.ToolsConfigInfo{}
 
-	if s.Config.ToolsUpgradePolicy {
-		configParams["tools.upgrade.policy"] = "upgradeAtPowerCycle"
+		if s.Config.ToolsSyncTime {
+			info.SyncTimeWithHost = &s.Config.ToolsSyncTime
+		}
+
+		if s.Config.ToolsUpgradePolicy {
+			info.ToolsUpgradePolicy = "UpgradeAtPowerCycle"
+		}
 	}
 
-	if len(configParams) > 0 {
+	if len(configParams) > 0 || info != nil {
```

Comment:
This condition here is a bit confusing because `AddConfigParams` checks again if they are empty/nil. 
I think you can move it to `AddConfigParams` before reconfiguring the VM like:

Reply:
You should have seen the first version....